### ISSUE TITLE
Authenticate calls

### ISSUE THREAD

**Issue by taromero** [takac/vim-spotifysearch#28]
When I try to use the plugin I'm receiving:

```
spotipy.client.SpotifyException: http status: 401, code:-1 - https://api.spotify.com/v1/search?q=somesong&type=track&limit=10&offset=0:
 No token provided
```

It seems like Spotify [removed unauthenticated calls to the API some months ago](https://developer.spotify.com/news-stories/2017/01/27/removing-unauthenticated-calls-to-the-web-api/), so in order for this to work it would need some sort of authentication.

**Comment by takac** [comment]
Yeah, I did notice this a couple months back :( This uses https://github.com/plamere/spotipy - Shouldn't be too hard to add in token support. Need to work out how best to pass in or retrieve a token.
